jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/DisplayMessageFragment.java

```diff
@@ -43,6 +43,8 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
      * @param newMessage The new message to be shown. */
     public void updateMessage(String newMessage) {
         message = newMessage;
+
+        // TODO: what if it is null?
```

Comment:
Comments are gone. But anyway, my point was this. If the `display_message`, for whatever weird reason (maybe some other developer came around and accidentally renamed an id) , is null, your app will seem broken. So what to do? One way would be to not check for null and let the app crash. I often consider that a better solution, because then the app will at least be restarted and then maybe work, and we are not leaving the user in doubt that things didn't work. As you will see in Kotlin with Optionals, we will often _force unwrap_ this kind of thing to say "this thing is nullable, but blow up if it is"

I would say a great solution is to force unwrap (Kotlin), and a __test__ that checks that this it not null, whenever the fragment has been created. Others may feel differently.